lib/containers/runtime.pm

```diff
@@ -0,0 +1,142 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test Docker’s btrfs storage driver features for image and container management
+# Maintainer: qac team <qa-c@suse.de>
+
+package containers::runtime;
+use Mojo::Base -base;
+use testapi;
+use Test::Assert 'assert_equals';
+
+has runtime => undef;
+
+sub _rt_assert_script {
+    my ($self, $cmd, @args) = @_;
+    assert_script_run($self->runtime . " " . $cmd, @args);
+}
+
+sub _rt_script_run {
+    my ($self, $cmd, @args) = @_;
+    return script_run($self->runtime . " " . $cmd, @args);
+}
+
+sub _rt_script_output {
+    my ($self, $cmd, @args) = @_;
+    return script_output($self->runtime . " " . $cmd, @args);
+}
+
+=head2 build
+
+Build a container after visit C<dockerfile_path>.
+C<dockerfile_path> is the directory with the Dockerfile.
+C<container_tag> will be the name of the container.
+
+=cut
+sub build {
+    my ($self, $dockerfile_path, $container_tag) = @_;
+    die 'wrong number of arguments' if @_ < 3;
+    assert_script_run "cd $dockerfile_path";
+    $self->_rt_assert_script("build -t $container_tag .");
+    record_info "$container_tag created";
+}
+
+=head2 up
+
+Run a container.
+If C<daemon> is enabled then container will run in the daemon mode. Otherwise will be in the 
+interactive mode.
+if C<cmd> found then it will execute the given command into the container.
+The container is always removed after exit.
+
+=cut
+sub up {
+    my ($self, $tag, %args) = @_;
+    die 'tag is required' unless $tag;
```

Comment:
why we can't start container without a tag ?

Reply:
tag is the name of the image that you run. $runtime->up('leap') translates to `docker run -it leap`. without the leap the command is not valid. do i miss something?

Reply:
> tag is the name of the image that you run. $runtime->up('leap') translates to `docker run -it leap`. without the leap the command is not valid. do i miss something?
 
in example which you gave `leap` is image name , tag is `:15.0` , `:latest` . and yes you can not run command without image name but you for sure can do it without a tag ( you will get `latest` in this case )

Reply:
Isn't it useful for versioning? Wouldn't we pull/start just the newest without `tag` ?  Just asking...

Reply:
from what I understand @b10n1k  just confused "tag" and "image name" . and we do pull latest without tag

Reply:
> > tag is the name of the image that you run. $runtime->up('leap') translates to `docker run -it leap`. without the leap the command is not valid. do i miss something?
> 
> in example which you gave `leap` is image name , tag is `:15.0` , `:latest` . and yes you can not run command without image name but you for sure can do it without a tag ( you will get `latest` in this case )

You are right. The variable name i use is confused with the actual tag. the `$tag` in this case can be the id, the name or the name+tag. i thought it would not matter.

Reply:
> from what I understand @b10n1k just confused "tag" and "image name" . and we do pull latest without tag

ah, yes. Got me as well :D 
@b10n1k just a make it explicit what we are starting. If the label represents image, what about `image_name` ?
If it should restart/respawn already existing container then named it accordingly. I know that labels are always a problem to ppl. Have the same problem all the time